test/verifyPath.test.js

```diff
@@ -0,0 +1,345 @@
+'use strict'
+
+const test = require('tap').test
+const Fastify = require('fastify')
+const fastifyCookie = require('@fastify/cookie')
+const fastifySession = require('..')
+const { DEFAULT_SECRET } = require('./util')
+
+test('should handle path properly /1', async (t) => {
+  t.plan(3)
+  const fastify = Fastify()
+
+  const options = {
+    secret: DEFAULT_SECRET,
+    cookie: { secure: false, path: '/' }
+  }
+  fastify.register(fastifyCookie)
+  fastify.register(fastifySession, options)
+  fastify.get('/', (request, reply) => {
+    request.session.foo = 'bar'
+    reply.send(200)
+  })
+  fastify.get('/check', (request, reply) => {
+    t.equal(request.session.foo, 'bar')
+    reply.send(200)
+  })
+  await fastify.listen({ port: 0 })
+  t.teardown(() => { fastify.close() })
```

Comment:
why do we `listen` and `close` when we use `inject`?

Reply:
I never thought about it. I just copy pasted other tests and thought I have to use listen to be able to do injects. If it is not the case, than maybe we can clean this up in follow up PR.

Reply:
Modified the new unit tests. But we should make a follow up PR removing it from the other places too, if applicable.